frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -217,6 +220,33 @@ export function InstantSearchProvider({
                   ? filter
                   : {};
 
+            Object.entries(filterObject).forEach(([attribute, value]) => {
+               if (!facets.includes(attribute)) {
+                  delete filterObject[attribute];
+                  return;
+               }
+               const widgetType = getFacetWidgetType(attribute);
+               switch (widgetType) {
+                  case FacetWidgetType.Menu: {
+                     if (isValidRefinementListValue(value)) {
+                        filterObject[attribute] = (value as string[])[0];
+                     } else if (typeof value !== 'string') {
```

Comment:
value should never be string here

Reply:
If it's a `Menu`, then the expected value is `string` (e.g. `[worksin]=Device Name`). However, in case the user visits a url with the refinement list array syntax (e.g. `[worksin][0]=Device Name`), we know they probably want the worksin filter, so that first `if` block will transform it into a string.

Reply:
*the expected **type** is `string`